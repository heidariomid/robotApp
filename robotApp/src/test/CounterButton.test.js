import React from 'react';
import {shallow} from 'enzyme';
import CounterButton from '../components/CounterButton';

describe('<CounterButton />', () => {
	it('renders components', () => {
		const colorMock = 'red';
		const wrapper = shallow(<CounterButton color={colorMock} />);
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('renders correctly', () => {
		const colorMock = 'red';
		const wrapper = shallow(<CounterButton color={colorMock} />);
		wrapper.find('[id=1]').simulate('click');
		wrapper.find('[id=1]').simulate('click');
		expect(wrapper.state()).toEqual({count: 2});
		expect(wrapper.props().color).toEqual('red');
	});
});
