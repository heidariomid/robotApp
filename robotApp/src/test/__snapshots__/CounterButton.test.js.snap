// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterButton /> renders components 1`] = `
"<button id={1} color=\\"red\\" onClick={[Function (anonymous)]}>
  Count: 
  0
</button>"
`;
